From c8e3d9e23d99f2f0f01bd5f367f6bce37d2f04b9 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Tue, 13 Aug 2019 00:14:54 +0200
Subject: [PATCH] test: store coredumps in journal

To make debugging much easier, especially for crashes in tests under
QEMU, let's store the entire coredump bundle in the systemd journal,
which is usually kept around by various CIs. Right now, we usually end
up with a journal, but without the coredump itself, which is pretty
useless.

(cherry picked from commit 215bffe1b8d7cb72fe9f72ed53682d52d5c2a9c5)
(cherry picked from commit cac1e15554b8e3f102e00ac802cff7100b1def22)

Resolves: #1836979
---
 test/test-functions | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/test-functions b/test/test-functions
index 3f1c327f3c..7c4230b078 100644
--- a/test/test-functions
+++ b/test/test-functions
@@ -420,6 +420,8 @@ install_systemd() {
 
     # enable debug logging in PID1
     echo LogLevel=debug >> $initdir/etc/systemd/system.conf
+    # store coredumps in journal
+    echo Storage=journal >> $initdir/etc/systemd/coredump.conf
 }
 
 get_ldpath() {
